### DISCUSSION TITLE
Using and linking with the modern zlib-ng API

### DISCUSSION THREAD

**Discussion by Diatrus** [nmoinvaz/minizip#573] [Q&A]
I was looking at adding minizip-ng to Procursus, and found it was linking the default zlib instead of the zlib-ng non-compat library. Looking into it, I also found an odd code-snippet in `mz_strm_zlib.c`:
```
#include "zlib.h"
#if defined(ZLIBNG_VERNUM) && !defined(ZLIB_COMPAT)
#  include "zlib-ng.h"
#endif
```
Isn't `ZLIBNG_VERNUM` defined in zlib-ng.h? Am I overlooking something?

**Comment by nmoinvaz** [comment]
Actually, I think it uses the zlib.h include which [contains](https://github.com/zlib-ng/zlib-ng/blob/5339a31cf9349a7b7e7ab8a5a507fda917f53f87/zlib.h#L50) the `ZLIBNG_VERNUM` #define to determine if it should include the zlib-ng API. This area might need to be reworked.

You can find information on how to build with zlib-ng here: https://github.com/zlib-ng/minizip-ng/tree/master/doc#zlib-configuration

**Reply by Diatrus** [reply]
I feel as though I have an alright grasp of how zlib-ng works. However, when disabling the compat API, zlib-ng doesn’t install a zlib.h header. I assume this is by design. The only way to get the expected behavior here would be to install the compat API.

**Reply by nmoinvaz** [reply]
In that case it is definitely an issue. Many things have changed in zlib-ng, my understanding also, since it was initially done.

**Reply by Diatrus** [reply]
I can look at making a PR in the next couple of days. Just wanted to make sure I wasn't missing something obvious.
